Propose an FAQ (#66)
commonhaus/foundation-draft: FAQ.md

---
status: draft
---

# Frequently Asked Questions (FAQs)

## Why create a new foundation?

There are already several foundations which host open source projects,
and each has its own focus and requirements.
The CF is a home for self-governing, code-centric open source projects.
Each open source project has its own unique needs and goals,
and not every foundation is the right fit for every project.

There are many critical application frameworks and libraries within the open source ecosystem that are not currently part of a foundation.
The CF is a new option for projects to consider when looking for a foundation to call home.

> Helping to strengthen the open source ecosystem as a whole by ensuring more projects are sustainable in the long term.

Reviewer: This section and the ones below mostly give reasons not to join other foundations, but no reason for joining CH... it's a bit weird, no?
As things are, it looks like not joining any foundation would be just as good a solution as joining CH.

Maybe we need at least a few arguments in favor of joining CH? I.e. "you'll get this", "you can hope for that", ...

Reviewer 2: I guess I agree with @yrodiere. And I'm not really convinced we even need to have a list of the "competition". Why not just remove all that and have a positive list of "differentiators", which doesn't need to be explicit about _who_ each point differentiates us from.

Author: Concerns were raised that not having an FAQ would have people wondering why not one of the other foundations.

If there's a consensus we don't need a deliberate FAQ answering these questions,  https://github.com/commonhaus/foundation-draft/blob/main/README.md covers the positives of the foundation.

Reviewer: That readme seems a bit vague and high-level to me, as I'd have expected something like "we provide infra" or "we provide access to lawyers" or "we provide spronsorship processes" or...

But anyway. I think having sections that explain "why not Apache/Eclipse/etc." makes sense, it's just that "Why create a new foundation?" would need some reason for the foundation to exist. Linking to the readme might provide that information?

Author: I've updated the "Why create a new foundation?" with links to the specific details in the README

Reviewer: Thanks @kenfinnigan , that looks good.

@ebullient Regarding the "why CH"... While I can understand the reluctance to over-promise, I think we agree there is a need to promise *something*.

Dreaming requires a fair amount of optimism, which I'm afraid is generally not my _forte_.

I can summarize what I got from previous readings and discussions though, by order of importance:

1. Independent, vendor-neutral IP ownership, with clear processes for both coordinating upstream projects and giving vendors controlled but fair influence. This is with the goal of fostering cooperation between multiple interested parties, as opposed to projects that seem "owned" by a company and may "repel" other companies.
2. Foundational rules: code of conduct, ...
3. Structure for small/starting projects that need it (a GitHub org, basically?)

Reviewer 3: Right - no need to make specific promises, but it would be nice to picture a clarer vision of what you'd hope this to become.
Some principles are nicely expressed already and I got a sense of why you're working on this, but I think it would help to make it a bit more concrete; perhaps some examples of what you'd hope to see coming? It's all sounding nice but it's also still quite vague.

---
status: draft
---

# Frequently Asked Questions (FAQs)

## Why create a new foundation?

Though there are several foundations for open source projects to choose to join,

> each has its own focus and requirements.

Reviewer 2: There are already several foundations which host open source projects, and each has its own focus and requirements.

The CF is a home for self-governing, code-centric open source projects.

> Each open source project has their own unique needs and goals,

Reviewer 2: has *its* own

and not every foundation is the right fit for every project.
The CF is a new option for projects to consider when looking for a foundation to call home.

There are many critical application frameworks and libraries within the open source ecosystem that are not currently part of a foundation.

> The CF is a place these projects may choose to call home,

Reviewer 2: repetitive

helping to strengthen the open source ecosystem as a whole by ensuring more projects are sustainable in the long term.

## Why not the Apache Software Foundation?

The Apache Software Foundation (ASF) has a long standing tradition of being a home for open source projects of all types,
and continues to be a great home for many.
As part of a project joining the ASF, it is required to run on the infrastructure that is provided for projects, including mailing lists,
and adhere to its processes.

For projects with well established infrastructure and processes, it's preferable for these to remain unchanged when joining a foundation.
Though the CF provides default processes for projects joining the foundation,
they're not mandatory when projects already have processes in place which achieve the same outcomes.

> All projects at the ASF producing software must be licensed under a version of the Apache Software License (ASL).

Reviewer 2: ~producing software~

At the CF, we provide projects a choice of several OSI approved licenses for their project.

## Why not the Eclipse Foundation?

The Eclipse Foundation (EF) has been around almost as long as the ASF, and is home to many open source projects as well.
In addition to open source projects, the EF oversees specification projects, particularly in relation to JakartaEE.
A key requirement for projects joining the EF is naming a project with the prefix _Eclipse_,
to reinforce brand recognition.

At the CF we strive to increase the brand of the project foremost, and the foundation secondarily.
The CF does not require project name changes when joining.

Open source projects at the CF may implement standards or specifications defined by other organizations,
such as the EF, but the CF will not oversee projects developing standards or specifications.
That is not where the CF will focus its efforts.

## Why not the Linux Foundation?

The Linux Foundation (LF) is home to many open source projects, including the Linux kernel itself.
Though the LF is a _neutral home_ for open source projects,
any project joining the LF is a very small part of a much larger organization.

> So much so, the LF itself is home to other foundations,

Reviewer 2: "home" is overused in this text, IMO.

---
status: draft
---

# Frequently Asked Questions (FAQs)

## Why create a new foundation?

There are already several foundations which host open source projects,
and each has its own focus and requirements.
The CF is a home for self-governing, code-centric open source projects.
Each open source project has its own unique needs and goals,
and not every foundation is the right fit for every project.

There are many critical application frameworks and libraries within the open source ecosystem that are not currently part of a foundation.
The CF is a new option for projects to consider when looking for a foundation to call home.

> Helping to strengthen the open source ecosystem as a whole by ensuring more projects are sustainable in the long term.

Reviewer 3: @gavinking said:

>  Why not just remove all that and have a positive list of "differentiators", which doesn't need to be explicit about who each point differentiates us from.

For the record, I had made the same suggestion over emails yesterday. I feel that calling out specific points of each foundation is unnecessary and could be abused to ignite unnecessary debate - we don't need this to be super specific like a laundry list of complaints.

Back to the specific wording of the FAQ: it feels like line 17 was meant to be a continuation of the previous line? Perhaps only separated by a comma?

Author: I've fixed the sentence continuation @Sanne, thanks

Information covering benefits of the CF are in the [README](README.md):

- [Core Distinctions of the CF](README.md#core-distinctions-of-cf)
- [CF Guiding Principles](README.md#our-guiding-principles)
- [Focus and Vision](README.md#cfs-focus-and-vision)
- [Community and Sponsorship](README.md#community-and-sponsorship)

## Why not the Apache Software Foundation?

The Apache Software Foundation (ASF) has a long standing tradition of being a home for open source projects of all types,
and continues to be a great home for many.
As part of a project joining the ASF, it is required to run on the infrastructure that is provided for projects, including mailing lists,
and adhere to its processes.

> For projects with well established infrastructure and processes, it's preferable for these to remain unchanged when joining a foundation.

Reviewer 3: ```suggestion
For projects with well established infrastructure and processes, it might be preferable for these to remain unchanged when joining a foundation.
```

Though the CF provides default processes for projects joining the foundation,

> they're not mandatory when projects already have processes in place which achieve the same outcomes.

Reviewer 3: ```suggestion
they're not mandatory.
```

Author: I'd be concerned taking out this statement implies a project could join without using the CF default processes and without any of its own processes

Reviewer 3: I see. I didn't know about CS wanting to impose anything in this area?

My concern with the original wording is that if one doesn't have "processes in place" then the "default processes" of the CF become mandatory. IMO that looks overly stronghanded, as in other areas the "default processes" seem more like suggestions (a word I've often seen is "templates").

I would have expected a newborn project would have some flexibility - obviously they'd need _some_ processes but they could tailor things a little? Or what if they initially want to use the CF "templates" but then feel the need to evolve things slightly? It would seem unfair if they are not allowed, while other projects which joined with different processes would be allowed to retain such flexibility. cc/ @ebullient

Reviewer 4: I only want to impose transparency to whatever it is that projects use to decide what they decide. I don't think we need to say "thou shalt have a process".

I agree with Sanne here. The requirement (for me) is that whatever it is is written down somehow (even if that comes down to the single maintainer making decisions based on what is best for the whole project given xyz factors).

All projects at the ASF must be licensed under a version of the Apache Software License (ASL).

> At the CF, we provide projects a choice of several OSI approved licenses for their project.

Reviewer 3: ```suggestion
At the CF, a wider choice of OSI approved licenses is welcome.
```

Author: Would "is available" be more accurate, as there will still be a defined set, at least initially?

Reviewer 3: Sure. But my version didn't imply that _every_ license is available either, so IMO both would be fine.

Reviewer 4: OSI-approved licenses are ok. I mention 4 because those are the ones we know. I don't have an exception list in mind.

I prefer Sanne's wording
